contrib/gcppubsub/pkg/adapter/adapter.go

```diff
@@ -98,23 +111,36 @@ func (a *Adapter) receiveMessage(ctx context.Context, m PubSubMessage) {
 	}
 }
 func (a *Adapter) postMessage(ctx context.Context, logger *zap.SugaredLogger, m PubSubMessage) error {
+	// Create the CloudEvent.
+	event := cloudevents.NewEvent(cloudevents.VersionV02)
+	event.SetID(m.ID())
+	event.SetTime(m.PublishTime())
+	event.SetDataContentType(*cloudevents.StringOfApplicationJSON())
+	event.SetSource(a.source)
+	event.SetData(m.Message())
+
 	// TODO: this will break when the upstream sender updates cloudevents versions.
 	// The correct thing to do would be to convert the message to a cloudevent if it is one.
 	et := sourcesv1alpha1.GcpPubSubSourceEventType
 	if override, ok := m.Message().Attributes[eventTypeOverrideKey]; ok {
 		et = override
 		logger.Infof("overriding the cloud event type with %q", et)
 	}
-
-	event := cloudevents.Event{
-		Context: cloudevents.EventContextV02{
-			Type:        et,
-			ID:          m.ID(),
-			Time:        &types.Timestamp{Time: m.PublishTime()},
-			Source:      *types.ParseURLRef(a.source),
-			ContentType: cloudevents.StringOfApplicationJSON(),
-		}.AsV02(),
-		Data: m.Message(),
+	event.SetType(et)
+
+	// If a transformer has been configured, then transform the message.
+	if a.transformer {
+		resp, err := a.transformerClient.Send(ctx, event)
+		if err != nil {
+			logger.Errorf("error transforming cloud event %q", event.ID())
+			return err
+		}
+		if resp != nil {
+			// Update the event with the transformed one.
+			event = *resp
+		} else {
+			logger.Warnf("cloud event %q was not transformed", event.ID())
```

Comment:
Should this return an error?

Reply:
I was leaving it as in case the transformer does not populate properly the response. If so, then we at least send the //google.pubsub.topic.publish event. As we might not control the transformer code, I'm assuming it can be wrong.
Do you prefer to return an error instead?

Reply:
I like the idea that the transformer can hide events. For example, events that are not on an allowed list or those that do not pass schema validation. I think our 'canonical' filtering of events is to return a 2XX with an empty body.

So I wouldn't return an error here, I would just `return nil`.